flutter/test/default_integrations_test.dart

```diff
@@ -283,6 +284,69 @@ void main() {
       expect(fixture.options.release, '__sentry_flutter _foo_bar_@1.2.3+789');
       expect(fixture.options.dist, '789');
     });
+
+    /// See the following issues:
+    /// - https://github.com/getsentry/sentry-dart/issues/410
+    /// - https://github.com/fluttercommunity/plus_plugins/issues/182
+    test('does not send Unicode NULL \\u0000 character in app name or version',
+        () async {
+      final loader = () {
+        return Future.value(PackageInfo(
+          // As per
+          // https://api.dart.dev/stable/2.12.4/dart-core/String-class.html
+          // this is how \u0000 is added to a string in dart
+          appName: 'sentry_flutter_example\u{0000}',
+          packageName: '',
+          version: '1.0.0\u{0000}',
+          buildNumber: '',
+        ));
+      };
+      await fixture
+          .getIntegration(loader: loader)
+          .call(MockHub(), fixture.options);
+
+      expect(fixture.options.release, 'sentry_flutter_example@1.0.0');
+    });
+
+    /// See the following issues:
+    /// - https://github.com/getsentry/sentry-dart/issues/410
+    /// - https://github.com/fluttercommunity/plus_plugins/issues/182
+    test(
+        'does not send Unicode NULL \\u0000 character in package name or build number',
+        () async {
+      final loader = () {
+        return Future.value(PackageInfo(
+          // As per
+          // https://api.dart.dev/stable/2.12.4/dart-core/String-class.html
+          // this is how \u0000 is added to a string in dart
+          appName: '',
+          packageName: 'sentry_flutter_example\u{0000}',
+          version: '',
+          buildNumber: '123\u{0000}',
+        ));
+      };
+      await fixture
+          .getIntegration(loader: loader)
+          .call(MockHub(), fixture.options);
+
+      expect(fixture.options.release, 'sentry_flutter_example+123');
```

Comment:
I think this is wrong, in this case, if there's no `version`, it should be `sentry_flutter_example@123`

Reply:
I'll confirm that, let's keep as it is for now

Reply:
this is actually not going to happen because pubspec must have a `version` or it defaults to 0.0.0 https://dart.dev/tools/pub/pubspec#version
its only possible because we've mocked the values